pkg/loader/load_test.go

```diff
@@ -378,6 +378,141 @@ func (s *batchManagerSuite) TestShouldExecAccumulatedDMLs(c *check.C) {
 	c.Assert(bm.txns, check.HasLen, 1)
 }
 
+type txnManagerSuite struct{}
+
+var _ = check.Suite(&txnManagerSuite{})
+
+func (s *txnManagerSuite) TestRunTxnManager(c *check.C) {
+	input := make(chan *Txn)
+	dmls := []*DML{
+		{Tp: UpdateDMLType},
+		{Tp: InsertDMLType},
+		{Tp: UpdateDMLType},
+	}
+	txn := &Txn{DMLs: dmls}
+	txnManager := newTxnManager(14, input)
+	output := txnManager.run()
+	// send 5 txns (size 3) to txnManager, the 5th txn should get blocked at cond.Wait()
+	for i := 0; i < 5; i++ {
+		select {
+		case input <- txn:
+		case <-time.After(50 * time.Microsecond):
+			c.Fatal("txnManager gets blocked while receiving txns")
+		}
+	}
+	c.Assert(output, check.HasLen, 4)
+	// Next txn should be blocked
+	select {
+	case input <- txn:
+		c.Fatal("txnManager doesn't block the txn when room is not enough")
+	case <-time.After(50 * time.Microsecond):
+	}
+	c.Assert(output, check.HasLen, 4)
+	// pick one txn from output channel
+	select {
+	case t := <-output:
+		txnManager.pop(t)
+		c.Assert(t, check.DeepEquals, txn)
+	case <-time.After(50 * time.Microsecond):
```

Comment:
No need to wait? Just use `default`, because if we can't get a txn at this point, something must be wrong.

Reply:
This place `default` can be used.